### PR TITLE
Topology optimization examples

### PR DESCRIPTION
**Pull request by artofscience**
Contributions to modules:
1. Allow for additional values on DOFs in assembly
3. Modules for stress calculation, vonmises stress and constraint aggregation (in ex_volume_stress.py)
4. Module for enforcing symmetry (in ex_flexure.py)
5. StaticCondensation module for use in kinetostatic formulation (ex_compliant_mechanism_kinetostatic.py)

Examples:
1. Compliant mechanism design using springs (Sigmund formulation)
6. Compliant mechanism design using kinetostatic approach (Koppen formulation)
7. Volume minimization under stress constraints
8. Compliance minimization under pure self-weight with optional stress constraints (different bc)
9. Optional stress constraints to design of flexures

Bug fixes:
1. Fixed sensitivity analysis of SoQ as to allow for Nonetype sensitivity input

Tests:
1. More tests for SoQ (test for Nonetype input sensitivity)
2. Test StaticCondensation module

Documentation:
1. Added documentation to multiple existing examples

### CODE REVIEW

**Review by aatmdelissen — CHANGES_REQUESTED** [review]

File: pymoto/__init__.py [hunk 1]
```diff
@@ -1,26 +1,25 @@
 __version__ = '1.1.0'
 
-from .core_objects import Signal, Module, Network, make_signals
-from .routines import finite_difference, minimize_oc, minimize_mma
-
```
[the remark this answers is not in the document]
Comment (artofscience) [reply pymoto/__init__.py:5]: good catch; it should not

File: pymoto/__init__.py [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
 from .modules.assembly import AssembleGeneral, AssembleStiffness, AssembleMass
+from .modules.autodiff import AutoMod
+from .modules.complex import MakeComplex, RealPart, ImagPart, ComplexNorm
 from .modules.filter import FilterConv, Filter, DensityFilter, OverhangFilter
+# Import from modules
+from .modules.generic import MathGeneral, EinSum, ConcatSignal
 from .modules.io import PlotDomain, PlotGraph, PlotIter, WriteToVTI
-from .modules.complex import MakeComplex, RealPart, ImagPart, ComplexNorm
-from .modules.autodiff import AutoMod
+from .modules.linalg import Inverse, LinSolve, EigenSolve, SystemOfEquations, StaticCondensation
```
Comment (artofscience) [inline pymoto/__init__.py:20]: ah static condensation module added

File: pymoto/modules/linalg.py [hunk 3]
```diff
@@ -13,6 +13,54 @@
...
+    Efficient multi-partition topology optimization.
+    Computer Methods in Applied Mechanics and Engineering, 393, 114829.
+    DOI: https://doi.org/10.1016/j.cma.2022.114829
+    """
+
+    def _prepare(self, free, main, **kwargs):
+        self.module_LinSolve = LinSolve([self.sig_in[0], Signal()], **kwargs)
+        self.module_LinSolve.use_lda_solver = False
+        self.m = main
+        self.f = free
```
[the remark this answers is not in the document]
Comment (artofscience) [reply pymoto/modules/linalg.py:47]: this is the minimum; partition consists of main and secondary, secondary is split in free and prescribed. so we already assume what is not free is prescribed here (zero prescribed value).

File: pymoto/modules/assembly.py [hunk 4]
```diff
@@ -34,7 +37,9 @@ class AssembleGeneral(Module):
         matrix_type (optional): The matrix type to construct. This is a constructor which must accept the arguments
           ``matrix_type((vals, (row_idx, col_idx)), shape=(n, n))``
     """
-    def _prepare(self, domain: DomainDefinition, element_matrix: np.ndarray, bc=None, bcdiagval=None, matrix_type=csc_matrix):
+
+    def _prepare(self, domain: DomainDefinition, element_matrix: np.ndarray, bc=None, bcdiagval=None,
+                 matrix_type=csc_matrix, add_values=None):
```
[the remark this answers is not in the document]
Comment (artofscience) [reply pymoto/modules/assembly.py:42]: added this as separate issue for future feature.

**Pull request merged**